Resources/doc/configuration.rst

```diff
@@ -993,52 +993,59 @@ can configure. The following block shows all possible configuration keys:
-                persistent="true"
-                multiple-active-result-sets="true" <!-- pdo_sqlsrv driver specific -->
-                pooled="true"                      <!-- Oracle specific (SERVER=POOLED) -->
-                protocol="TCPIP"                   <!-- IBM DB2 specific (PROTOCOL) -->
-                server="my_database_server"        <!-- SQL Anywhere specific (ServerName) -->
-                service="true"                     <!-- Oracle specific (SERVICE_NAME instead of SID) -->
-                servicename="MyOracleServiceName"  <!-- Oracle specific (SERVICE_NAME) -->
-                sessionMode"2"                     <!-- oci8 driver specific (session_mode) -->
-                default_dbname="database"          <!-- PostgreSQL specific (default_dbname) -->
-                sslmode="require"                  <!-- PostgreSQL specific (LIBPQ-CONNECT-SSLMODE) -->
-                sslrootcert="postgresql-ca.pem"    <!-- PostgreSQL specific (LIBPQ-CONNECT-SSLROOTCERT) -->
-                sslcert="postgresql-cert.pem"      <!-- PostgreSQL specific (LIBPQ-CONNECT-SSLCERT) -->
-                sslkey="postgresql-key.pem"        <!-- PostgreSQL specific (LIBPQ-CONNECT-SSLKEY) -->
-                sslcrl="postgresql.crl"            <!-- PostgreSQL specific (LIBPQ-CONNECT-SSLCRL) -->
-                wrapper-class="MyDoctrineDbalConnectionWrapper"
-                charset="UTF8"
-                logging="%kernel.debug%"
-                platform-service="MyOwnDatabasePlatformService"
-                auto-commit="false"
-                schema-filter="^sf2_"
-            >
-                <doctrine:option key="foo">bar</doctrine:option>
-                <doctrine:mapping-type name="enum">string</doctrine:mapping-type>
-                <doctrine:default-table-option name="charset">utf8</doctrine:default-table-option>
-                <doctrine:default-table-option name="collate">utf8_unicode_ci</doctrine:default-table-option>
-                <doctrine:default-table-option name="engine">InnoDB</doctrine:default-table-option>
-                <doctrine:type name="custom">Acme\HelloBundle\MyCustomType</doctrine:type>
-            </doctrine:dbal>
-        </doctrine:config>
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:doctrine="http://symfony.com/schema/dic/doctrine"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/doctrine
+                http://symfony.com/schema/dic/doctrine/doctrine-1.0.xsd">
+
+            <doctrine:config>
+                <doctrine:dbal
+                    name="default"
+                    url="mysql://user:secret@localhost:1234/otherdatabase" <!-- this would override the values below -->
```

Comment:
these inline comments are the actual issue, we cannot use them in XML inside tags

Reply:
I see. It makes sense now. Any idea to solve this? Thanks.

Reply:
Good catch. I thought this was about inconsistent XML, but the Doctrine docs are having issues too with them: https://www.doctrine-project.org/projects/doctrine-bundle/en/latest/configuration.html#doctrine-dbal-configuration

Reply:
In the Symfony docs we usually add comments above the tag where we want to comment some attributes. That might be a bit too much here But maybe not all of the comments are really necessary.

Reply:
So, what should we do? Keep the comments and lose highlighting (not a big deal) or move the comments at the beginning (it may look crowded, as Christian said).

Reply:
Maybe provide several examples with comments at the beginning? That would help grouping things that go together (url vs non-url, RDMBS-specific settings, etc…)

Reply:
I extracted the comments. @xabbuh can you please take another look?